test/router.js

```diff
@@ -1129,6 +1129,31 @@ describe('Router', function () {
       .expect(200, 'saw GET /bar', done)
     })
   })
+
+  describe('req.matchedRoutes', function () {
```

Comment:
One test doesn't seem adequate for this feature. Here are a few more tests cases I can think of off-hand:

* Check that it works for regexp-defined routes
* Check that is works as expected with sub routers

Reply:
Yeah, I tend to agree. I'll try to add more tests.

Reply:
Updated with some more tests. Let me know if that's what you had in mind. They don't totally match the style of the existing code, so let me know if it should be changed.

Reply:
Nice! Can you add a test for what req.matchedRoutes looks like in the top router after exiting the sub router as well?

Reply:
@dougwilson I think that's failing. I added a test that looks like this:
```js
var router = new Router()
var fooRouter = new Router()
var server = createServer(router)
var matchedFooRoutes
var matchedBarRoutes

router.use('/foo', function (req, res, next) {
    matchedFooRoutes = req.matchedRoutes
    next()
}, fooRouter)
fooRouter.get('/bar', function (req, res, next) {
    matchedBarRoutes = req.matchedRoutes
    next()
})
router.use(saw)

request(server)
    .get('/foo/bar')
    .expect(200, 'saw GET /foo/bar', function (err, res) {
       // matchedFooRoutes is [ '/foo', '/foo', '/bar' ] instead
        assert.deepEqual(matchedFooRoutes, ['/foo']) 
        done(err)
    })
```

I'll see if I can fix it.

Reply:
Fixed with my most recent commit, but not totally sure that's the right way to go. Let me know what you think.